MC/config/PWGHF/external/generator/generator_pythia8_gaptriggered_hf.C

```diff
@@ -2,37 +2,63 @@
 #include "Generators/GeneratorPythia8.h"
 #include "Pythia8/Pythia.h"
 #include "TRandom.h"
+#include <fairlogger/Logger.h>
 
 #include <string>
 #include <vector>
 
 using namespace Pythia8;
 
-class GeneratorPythia8GapTriggeredHF : public o2::eventgen::GeneratorPythia8 {
+class GeneratorPythia8GapTriggeredHF : public o2::eventgen::GeneratorPythia8
+{
 public:
-
   /// default constructor
   GeneratorPythia8GapTriggeredHF() = default;
 
   /// constructor
-  GeneratorPythia8GapTriggeredHF(int inputTriggerRatio = 5, std::vector<int> quarkPdgList = {4}, bool doAltInjection = false) {
+  GeneratorPythia8GapTriggeredHF(int inputTriggerRatio = 5, std::vector<int> quarkPdgList = {}, std::vector<int> hadronPdgList = {})
+  {
 
     mGeneratedEvents = 0;
-    mHadronPdg = 0; // unless differently set, we do not trigger on specific hadron species
-    mQuarkPdg = quarkPdgList[0];
     mInverseTriggerRatio = inputTriggerRatio;
     mQuarkRapidityMin = -1.5;
     mQuarkRapidityMax = -1.5;
     mHadRapidityMin = -1.5;
     mHadRapidityMax = -1.5;
-    mDoNoQuarkTrigger = false;
-    mDoAltInjection = doAltInjection;
+    mDoNoQuarkTrigger = quarkPdgList.size() == 0;
     mQuarkPdgList = quarkPdgList;
+    mHadronPdgList = hadronPdgList;
+    Print();
```

Comment:
I would add here a protection that does not allow to have `size > 1` for both parton and hadron lists, because otherwise one gets always the same hadron associated to the same hadron in the current logic and there is no diagonal case (i.e. in case of 2 partons and 2 hadrons, `parton0` will always be triggered with `hadron0` and `parton1` with `hadron1` but there is no `parton0 - hadron1` and `parton1 - hadron0` generation.